src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -879,7 +879,13 @@ def _find_using_common_tenant(self, access_token, resource):
                 # because user creds went through the 'common' tenant, the error here must be
                 # tenant specific, like the account was disabled. For such errors, we will continue
                 # with other tenants.
-                logger.warning("Failed to authenticate '%s' due to error '%s'", t, ex)
+                msg = (getattr(ex, 'error_response', None) or {}).get('error_description') or ''
```

Comment:
why not use `getattr(ex, 'error_response', {})` and `get('error_description', '')` ?

Reply:
Good question. I previously have exactly the same question as you. Because `error_response` can be `None`, instead of not-exist. In that case, `getattr(ex, 'error_response', {})` returns `None` and cause following `.get` to fail.

Reply:
Same for `get('error_description', '')`, which may return `None` if `error_description` exists, but the value is `None`. `get('error_description') or ''` guarantees the returned value to be at least an empty string.

Reply:
got it. make sense.